tests/functional/servers/addon.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Server = require('./server.js');
+const constants = require('../constants.js');
+const fs = require('fs');
+const path = require('path');
+
+const ADDON_PATH = './tests/functional/addons';
+
+let server = null;
+module.exports = {
+  start() {
+    let scenarios = {};
+
+    const dirs = fs.readdirSync(ADDON_PATH);
+    for (const dir of dirs) {
+      const addonPath = path.join(ADDON_PATH, dir);
+      const stat = fs.statSync(addonPath);
+      if (!stat.isDirectory()) {
+        continue;
+      }
+
+      scenarios = {...scenarios, ...this.createScenario(dir, addonPath)};
+    }
+
+    const endpoints = {
+      GETs: scenarios,
+      POSTs: {},
+      DELETEs: {},
+    };
+
+    server = new Server('Addon', constants.ADDON_PORT, endpoints);
+
+    return constants.ADDON_PORT;
+  },
+
+  stop() {
+    server.stop();
+  },
+
+  get overrideEndpoints() {
+    return server.overrideEndpoints;
+  },
+
+  set overrideEndpoints(value) {
+    server.overrideEndpoints = value;
+  },
+
+  throwExceptionsIfAny() {
+    server.throwExceptionsIfAny();
+  },
+
+  createScenario(scenario, addonPath) {
+    const generatedPath = path.join(addonPath, 'generated');
+    if (fs.existsSync(generatedPath)) {
+      return this.createScenarioGenerated(scenario, generatedPath);
+    }
+
+    const manifestPath = path.join(addonPath, 'manifest.json');
+    if (!fs.existsSync(manifestPath)) {
+      throw new Error(
+          `No generated and not manifest file! ${manifestPath} should exist!`);
+    }
+
+    const obj = {};
+
+    obj[`/${scenario}/manifest.json`] = {
+      status: 200,
+      bodyRaw: fs.readFileSync(manifestPath),
+    };
+    obj[`/${scenario}/manifest.json.sig`] = {
+      status: 404,
+      bodyRaw: '',
+    };
+
+    return obj;
+  },
+
+  createScenarioGenerated(scenario, generatedPath) {
```

Comment:
If this function is not supposed to be called outside of this module, let's leave it our of the exported object.